src/model.ts

```diff
@@ -7,67 +7,123 @@ import { IContentFieldFilter, IContentSearchRequest } from '@esri/hub-search';
+  'type', // used for the dataset landing page URL
+  'slug', // used for the dataset landing page URL
+  'access', // used for detecting proxied csv's
+  'size', // used for detecting proxied csv's
+  'licenseInfo', // required for license resolution
+  'structuredLicense', // required for license resolution
+];
+
+// additional fields due to dataset enrichment
+const ADDON_FIELDS = [
+  'hubLandingPage',
+  'accessUrlCSV',
+  'isLayer',
+  'accessUrlKML',
+  'accessUrlShapeFile',
+  'accessUrlWFS',
+  'accessUrlWMS',
+  'accessUrlGeoJSON',
+  'license'
+];
+
+type HubApiRequest = {
+  res?: {
+    locals: {
+      searchRequestBody?: IContentSearchRequest,
+      siteModel?: IModel,
+      arcgisPortal?: string
+      siteIdentifier?: string,
+    }
+  }
+  query?: {
+    limit?: number
+  }
+}
 
 export class HubApiModel {
 
-  async getStream (request: Request) {
-    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
-    this.preprocessSearchRequest(searchRequest);
+  async getStream(request: Request) {
+   
+    const { res: { locals: { searchRequestBody, siteModel, arcgisPortal, siteIdentifier } }, query: { limit } }: HubApiRequest = request;
+    const hubSiteModel = siteModel || await this.fetchHubSiteModel(siteIdentifier, this.getRequestOptions(arcgisPortal));
+    this.preprocessSearchRequest(searchRequestBody);
 
-    if (searchRequest.options.fields) {
+    if (searchRequestBody.options.fields) {
+      searchRequestBody.options.fields = this.getValidHubApiFields(searchRequestBody.options.fields);
       const validFields: string[] = await hubApiRequest('/fields');
-      this.validateFields(searchRequest, validFields);
+      this.validateFields(searchRequestBody, validFields);
     }
 
     // Only fetch site if site is provided and either group or orgid is undefined
     if (
-      searchRequest.options.site &&
+      searchRequestBody.options.site &&
       (
-        !this.scopedFieldValueIsValid(searchRequest.filter.group) || 
-        !this.scopedFieldValueIsValid(searchRequest.filter.orgid)
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.group) ||
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.orgid)
       )
     ) {
-      const siteCatalog = await this.getSiteCatalog(searchRequest, searchRequest.options.site);
-      if (!this.scopedFieldValueIsValid(searchRequest.filter.group)) {
-        searchRequest.filter.group = siteCatalog.groups;
+      const siteCatalog = await this.getSiteCatalog(searchRequestBody, searchRequestBody.options.site);
+      if (!this.scopedFieldValueIsValid(searchRequestBody.filter.group)) {
+        searchRequestBody.filter.group = siteCatalog.groups;
       }
-      if (!this.scopedFieldValueIsValid(searchRequest.filter.orgid)) {
-        searchRequest.filter.orgid = siteCatalog.orgId;
+      if (!this.scopedFieldValueIsValid(searchRequestBody.filter.orgid)) {
+        searchRequestBody.filter.orgid = siteCatalog.orgId;
```

Comment:
move contents of block to a helper.

Reply:
Extracted the contents of the block to a helper function called `addGroupAndOrgId(searchRequestBody)`